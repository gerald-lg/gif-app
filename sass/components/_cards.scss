@use '../abstract/colors' as *;

.card {
    background-color: $white;
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    height: 0%;  
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
    width: 400px;
    height: auto;
}
  
.card img {
    object-fit: cover;
    width: 100%;
    max-height: 350px;
}

.card:hover{
    -ms-transform: scale(1.1);
    transform: scale(1.1);
    transition: 0.5s ease;
}

.card p {
    text-align: center;
    font-size: 1.2rem;
    background-color: $white;
    margin-top: 1.5em;
}