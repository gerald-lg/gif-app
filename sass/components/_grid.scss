@use '../abstract/colors' as *;

 .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;

    span.loader{
        margin: 50px;
    }
}
.title__center{
    display: flex;
    justify-content: center;

    text-transform: uppercase;
    margin-bottom: 15px;
}

