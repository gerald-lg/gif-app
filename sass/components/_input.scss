


.custom__input{
    padding: 0.8em;
    outline: none;
    border: 1px solid #EEE;
    background-color: transparent;
    border-radius: 5px;
    width: 100%;
    color: #fff;
}