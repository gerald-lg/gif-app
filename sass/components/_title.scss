@use '../abstract/mixins' as *;
@use '../abstract/typography' as *;

.waviy {
    position: relative;
    font-size: 60px;
    font-weight: 700;
    font-family: $custom-font;

    @include breakpoint(md, max){
      font-size: 40px;
    }
  }
  .waviy span {
    position: relative;
    display: inline-block;
    color: #fff;
    text-transform: uppercase;
    animation: waviy 1s infinite;
    animation-delay: calc(.1s * var(--i));
    
  }
  @keyframes waviy {
    0%,40%,100% {
      transform: translateY(0)
    }
    20% {
      transform: translateY(-20px)
    }
  }

