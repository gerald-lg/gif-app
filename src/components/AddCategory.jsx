import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( { target } ) => {
        setInputValue(target.value);
    };

    const onSubmit = ( (event) => {
        event.preventDefault();

        if( inputValue.trim().length <= 1) return; 
        onNewCategory( inputValue.trim() );
        cleanValue();
    });

    
   
    const cleanValue = () => {
        setInputValue('');
    }

    return (
        <form method="post" onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text" 
                placeholder="Buscar gifs" 
                value={inputValue} 
                onChange={ onInputChange }
                className="custom__input"
                />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}