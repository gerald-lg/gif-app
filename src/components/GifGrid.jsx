import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( { category } ) => {



   const {images, isLoading} = useFetchGifs( category );
    
    
    return (
        <>
            {
                //and logico
                isLoading && ( <div className="card-grid"><span className="loader"></span></div> )
            }

            <h1 className='title__center'>{ category }</h1>
            
            <div className="card-grid">
                { images.map( (image) => (
                    // <li key={ id }> { title } </li>
                    <GifItem key={ image.id } 
                        // tecnica que se utiliza cuando existen muchas props, se esparcen las props
                        { ...image }
                    />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired,
}
