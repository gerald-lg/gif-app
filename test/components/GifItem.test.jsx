import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas sobre el componente GifItem', () => {
    
    const title = 'Hola';
    const url = 'cualquierurl';
    
    test('Debe hacer match con el snapshot', () => {
        
        const { container } = render(<GifItem title={title} url={url} />);
        
        expect(container).toMatchSnapshot();

    })
 })