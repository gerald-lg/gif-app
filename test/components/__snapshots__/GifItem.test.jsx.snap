// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="cualquierurl"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
